pkg/apiserver/registry/system/supportbundle/rest_test.go

```diff
@@ -174,3 +181,110 @@ func TestControllerStorage(t *testing.T) {
 		Status:     system.SupportBundleStatusNone,
 	}, object)
 }
+
+func TestNewAgentStorage(t *testing.T) {
+	defaultFS = afero.NewMemMapFs()
+	defaultExecutor = new(testExec)
+	defer func() {
+		defaultFS = afero.NewOsFs()
+		defaultExecutor = exec.New()
+	}()
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	fakeOVSCtl := ovsctltest.NewMockOVSCtlClient(ctrl)
+	fakeAgentQuerier := agentqueriertest.NewMockAgentQuerier(ctrl)
+	fakeNetworkPolicyQuerier := queriertest.NewMockAgentNetworkPolicyInfoQuerier(ctrl)
+	fakeIPTables := iptablestest.NewMockInterface(ctrl)
+	fakeOVSCtl.EXPECT().DumpFlows()
+	fakeOVSCtl.EXPECT().DumpPortsDesc()
+	fakeNetworkPolicyQuerier.EXPECT().GetAddressGroups()
+	fakeNetworkPolicyQuerier.EXPECT().GetAppliedToGroups()
+	fakeNetworkPolicyQuerier.EXPECT().GetNetworkPolicies(&querier.NetworkPolicyQueryFilter{})
+	fakeAgentQuerier.EXPECT().GetAgentInfo(&clusterinformationv1beta1.AntreaAgentInfo{}, false)
+	fakeIPTables.EXPECT().Save()
+	storage := NewAgentStorage(fakeOVSCtl, fakeAgentQuerier, fakeNetworkPolicyQuerier, fakeIPTables)
+	_, err := storage.SupportBundle.Create(context.TODO(), &system.SupportBundle{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: modeAgent,
+		},
+		Status: system.SupportBundleStatusNone,
+		Since:  "-1h",
+	}, nil, nil)
+	require.NoError(t, err)
+
+	var collectedBundle *system.SupportBundle
+	assert.Eventually(t, func() bool {
```

Comment:
this is neat. Is this new?
should we switch to `assert.Eventually` for all new unit tests instead of `wait.Poll`?

Reply:
Added in 1.4.0. Yes, I think it can replace wait.Poll in tests.